src/Salt/SiteBundle/Entity/Comment.php

```diff
@@ -35,7 +37,7 @@ class Comment
      * @var Comment
      *
      * @ORM\ManyToOne(targetEntity="\Salt\SiteBundle\Entity\Comment")
-     * @ORM\JoinColumn(nullable=false, onDelete="CASCADE")
+     * @ORM\JoinColumn(nullable=true, onDelete="CASCADE")
```

Comment:
Is there a special reason to make the user nullable? Also notice that you didn't add the migration to make it nullable.